### ISSUE TITLE
[Bug]: Construting sandbag bunker on garrison counts as a static weapon

### ISSUE THREAD

**Issue by awpexx** [official-antistasi-community/A3-Antistasi#2642]
### Describe the bug

This bug involves the construction of a sandbag bunker on a friendly garrison counting as a static weapon. This will cause 1 rifleman to be confused, as it will not "teleport" to it's random predesignated position in the garrison, causing it to be static near the flag, or aimlessly wander around.
My speculation on why the rifleman acts this way, is because it counts the bunker as a static weapon (machine gun, etc.), but since the bunker has no "seats" for the rifleman to mount in, it acts the way it does.
It would be nice if this bug was resolved so that the rifleman acts the way it is intended too.

### How to reproduce

1. Be in a friendly garrisoned position.
2. Have at least 1 rifleman
3. Construct a sandbag bunker
4. See in the "map info" section that the sandbag bunker counts as a static weapon
5. Reload the garrison (eg. travel more than 1km far away from it)
6. Watch as the rifleman exhibits the behavior described

### Version

3.1.0. but i've experienced it on version 2.5.5.

### Have you altered the code?

No

### Map

Livonia

### What server?

LAN Hosted

### Time bug occured (Server time/UTC)

not relevant

### Mods

```html
3CB Factions 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1673456286
ace 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=463939057
ACE Compat - RHS AFRF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=773131200
ACE Compat - RHS USAF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=773125288
ACE Compat - RHS: GREF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=884966711
ACE Compat - RHS: SAF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=2174495332
Antistasi - The Mod 	Steam 	https://steamcommunity.com/sharedfiles/filedetails/?id=2867537125
Blastcore Edited (standalone version) 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=767380317
CBA_A3 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=450814997
DUI - Squad Radar 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1638341685
Enhanced Movement 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=333310405
Enhanced Movement Rework 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=2034363662
Gruppe Adler Trenches 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1224892496
JSRS SOUNDMOD 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=861133494
JSRS SOUNDMOD - RHS AFRF Mod Pack Sound Support 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=945476727
JSRS SOUNDMOD - RHS GREF Mod Pack Sound Support 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1180534892
JSRS SOUNDMOD - RHS SAF Mod Pack Support 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1486541773
JSRS SOUNDMOD - RHS USAF Mod Pack Sound Support 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1180533757
Looting Enhanced 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=2479270597
RHSAFRF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843425103
RHSGREF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843593391
RHSSAF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843632231
RHSUSAF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843577117
VET_Unflipping 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1703187116
```

**Comment by jaj22** [comment]
Confirmed. The updateRebelStatics function uses staticsToSave as the source data, and some constructed objects are stored in staticsToSave.

Overdue a full rework of garrison storage/management logic, and at least a rework to store garrison vehicles persistently in garrison data. If neither of these are done by the next version then I might fix this directly.

**Comment by jaj22** [comment]
Fixed in #2690
